syntax = "proto3";
package contacts;
import "google/protobuf/any.proto";

message Address{
    string home = 1;
    string work = 2;
}

// 增加电话类型
enum PhoneType{
    UNKNOWN = 0;
    MOBILE_PHONE = 1;
    LANDLINE = 2;
}

// 电话信息
message Phone{
    string number = 1;
    PhoneType type = 2;
}

// 联系人信息
message People{
    string name = 1;
    int32 age = 2;
    repeated Phone phone = 3;
    google.protobuf.Any date = 4;

    // 使用oneof时，内部的字段的字段编号必须接着之前的字段编号进行
    // 类似于C++的union联合体
    oneof other_contact{
        string qq = 5;
        string wechat = 6;
    }
}

// 通讯录
message Contacts{
    repeated People people = 1;
}